pgxn/neon/neon_walreader.c

```diff
@@ -36,13 +36,35 @@
 
 static NeonWALReadResult NeonWALReadRemote(NeonWALReader *state, char *buf, XLogRecPtr startptr, Size count, TimeLineID tli);
 static NeonWALReadResult NeonWALReaderReadMsg(NeonWALReader *state);
-static void NeonWALReaderResetRemote(NeonWALReader *state);
 static bool NeonWALReadLocal(NeonWALReader *state, char *buf, XLogRecPtr startptr, Size count, TimeLineID tli);
-static bool neon_wal_segment_open(NeonWALReader *state, XLogSegNo nextSegNo, TimeLineID *tli_p);
-static void neon_wal_segment_close(NeonWALReader *state);
 static bool is_wal_segment_exists(XLogSegNo segno, int segsize,
 								  TimeLineID tli);
 
+/*
+ * It's kind of cursed to define this here, but we can't define it in
+ * walproposer.c because that makes the simulation tester try to pull it in.
+ */
+SafekeeperShmemState *
+GetDonorShmem(XLogRecPtr *donor_lsn)
```

Comment:
This function is names `GetDonorShmem` while actually it is used to get donor LSN and returned donor's shame state  is just use to check if there is some donor.